API/src/core/ApiResponse.ts

```diff
@@ -0,0 +1,120 @@
+import { Response } from 'express';
+
+enum StatusCode {
+    SUCCESS = '10000',
+    FAILURE = '10001',
+    RETRY = '10002',
+    INVALID_ACCESS_TOKEN = '10003',
+}
+
+enum ResponseStatus {
+    SUCCESS = 200,
+    BAD_REQUEST = 400,
+    UNAUTHORIZED = 401,
+    FORBIDDEN = 403,
+    NOT_FOUND = 404,
+    INTERNAL_ERROR = 500,
+}
+
+export abstract class ApiResponse {
+    constructor(
+        protected statusCode: StatusCode,
+        protected status: ResponseStatus,
+        protected message: string,
+    ) {}
+
+    /*
+     * @function        send
+     * @desc            given the Response object, create Response with given data this object contains
+     *
+     * @params          res {Response}  response object from controller
+     *
+     * @return          Response object filled with header status, data and other data needed for the response
+     */
+    public send(res: Response): Response {
+        return this.prepare(res, this);
+    }
+
+    /*
+     * @function        prepare
+     * @desc            prepare response with given status and data respone
+     * @params          res {Response} response object
+     * @params          response {extends ApiResponse} the response data in response body
+     *
+     * @return          response object with given status in header and jsonified T's data in body
+     */
+    protected prepare<T extends ApiResponse>(
+        res: Response,
+        response: T,
+    ): Response {
+        return res.status(this.status).json(ApiResponse.sanitize(response));
+    }
+
+    /*
+     * @function        sanitize
+     * @desc            prepare json object for the response
+     *                  remove all unneeded fields and functions
+     * @params          T {extends ApiResponse} subclass of ApiResponse contatining data for response.
+     *
+     * @returns         an object contains: statusCode, data (if subclass has the field)
+     */
+    private static sanitize<T extends ApiResponse>(response: T): T {
```

Comment:
one thing to add (or at least think about), is that the API will take a global parameter of the response format to return (ie json, csv, text,.....) so having something that processes the different format types in the response will be needed

Reply:
okay, I see. I didnt think about other format for the response. I'll change it.